File: src/Entity/Virement.php
<?php

namespace App\Entity;

use App\Repository\VirementRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: VirementRepository::class)]
class Virement
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(inversedBy: 'virements')]
    private ?Compte $id_compte_emetteur = null;

    #[ORM\ManyToOne(inversedBy: 'virements_recus')]
    private ?Compte $id_compte_beneficiaire = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $cin_emetteur = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $photo_cin = null;

    #[ORM\Column]
    private ?float $montant = null;

    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    private ?\DateTimeInterface $date_emission = null;

    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    private ?\DateTimeInterface $date_approbation = null;

    #[ORM\Column(length: 255)]
    private ?string $etat = null;

    #[ORM\ManyToOne(inversedBy: 'virements')]
    private ?TypeVirement $type = null;

    public function __construct()
    {
        $this->date_emission = new \DateTime();
        $this->etat = "Succès";
    }
    
    public function getType(): ?TypeVirement
    {
        return $this->type;
    }
    public function setType(?TypeVirement $type): static
    {
        $this->type = $type;

        return $this;
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getIdCompteEmetteur(): ?Compte
    {
        return $this->id_compte_emetteur;
    }

    public function setIdCompteEmetteur(?Compte $id_compte_emetteur): static
    {
        $this->id_compte_emetteur = $id_compte_emetteur;

        return $this;
    }

    public function getIdCompteBeneficiaire(): ?Compte
    {
        return $this->id_compte_beneficiaire;
    }

    public function setIdCompteBeneficiaire(?Compte $id_compte_beneficiaire): static
    {
        $this->id_compte_beneficiaire = $id_compte_beneficiaire;

        return $this;
    }

    public function getCinEmetteur(): ?string
    {
        return $this->cin_emetteur;
    }

    public function setCinEmetteur(?string $cin_emetteur): static
    {
        $this->cin_emetteur = $cin_emetteur;

        return $this;
    }

    public function getPhotoCin(): ?string
    {
        return $this->photo_cin;
    }

    public function setPhotoCin(?string $photo_cin): static
    {
        $this->photo_cin = $photo_cin;

        return $this;
    }

    public function getMontant(): ?float
    {
        return $this->montant;
    }

    public function setMontant(float $montant): static
    {
        $this->montant = $montant;

        return $this;
    }

    public function getDateEmission(): ?\DateTimeInterface
    {
        return $this->date_emission;
    }

    public function setDateEmission(\DateTimeInterface $date_emission): static
    {
        $this->date_emission = $date_emission;

        return $this;
    }

    public function getDateApprobation(): ?\DateTimeInterface
    {
        return $this->date_approbation;
    }

    public function setDateApprobation(\DateTimeInterface $date_approbation): static
    {
        $this->date_approbation = $date_approbation;

        return $this;
    }

    public function getEtat(): ?string
    {
        return $this->etat;
    }

    public function setEtat(string $etat): static
    {
        $this->etat = $etat;

        return $this;
    }


   
}

File: src/main/java/com/inventorymanagementsystem/entity/Virement.java
package com.inventorymanagementsystem.entity;

import javafx.beans.property.SimpleBooleanProperty;

import java.util.Objects;

public class Virement implements Comparable<Virement> {

    private Integer id;
     private Compte idCompteEmetteur;
    private Compte idCompteBeneficiaire;
    private String cinEmetteur;
    private String photoCin;
    private Double montant;
    private java.util.Date dateEmission;
    private java.util.Date dateApprobation;
    private String etat;
    private TypeVirement type;

    // Constructor
    public Virement() {
        this.dateEmission = new java.util.Date();
        this.etat = "Succès";
    }

     public Virement(Integer id, Compte idCompteEmetteur, Compte idCompteBeneficiaire, String cinEmetteur, String photoCin, Double montant, java.util.Date dateEmission, java.util.Date dateApprobation, String etat, TypeVirement type) {
         this.id = id;
         this.idCompteEmetteur = idCompteEmetteur;
         this.idCompteBeneficiaire = idCompteBeneficiaire;
         this.cinEmetteur = cinEmetteur;
         this.photoCin = photoCin;
         this.montant = montant;
         this.dateEmission = dateEmission;
         this.dateApprobation = dateApprobation;
         this.etat = etat;
         this.type = type;
     }
    public Virement(Integer id,  String cinEmetteur, String photoCin, Double montant, java.util.Date dateEmission, java.util.Date dateApprobation, String etat, TypeVirement type) {
        this.id = id;

        this.cinEmetteur = cinEmetteur;
        this.photoCin = photoCin;
        this.montant = montant;
        this.dateEmission = dateEmission;
        this.dateApprobation = dateApprobation;
        this.etat = etat;
        this.type = type;
    }

    // Getters and Setters
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

   /* public Compte getIdCompteEmetteur() {
        return idCompteEmetteur;
    }

    public void setIdCompteEmetteur(Compte idCompteEmetteur) {
        this.idCompteEmetteur = idCompteEmetteur;
    }

    public Compte getIdCompteBeneficiaire() {
        return idCompteBeneficiaire;
    }

    public void setIdCompteBeneficiaire(Compte idCompteBeneficiaire) {
        this.idCompteBeneficiaire = idCompteBeneficiaire;
    }*/

    public String getCinEmetteur() {
        return cinEmetteur;
    }

    public void setCinEmetteur(String cinEmetteur) {
        this.cinEmetteur = cinEmetteur;
    }

    public String getPhotoCin() {
        return photoCin;
    }

    public void setPhotoCin(String photoCin) {
        this.photoCin = photoCin;
    }

    public Double getMontant() {
        return montant;
    }

    public void setMontant(Double montant) {
        this.montant = montant;
    }

    public java.util.Date getDateEmission() {
        return dateEmission;
    }

    public void setDateEmission(java.util.Date dateEmission) {
        this.dateEmission = dateEmission;
    }

    public java.util.Date getDateApprobation() {
        return dateApprobation;
    }

    public void setDateApprobation(java.util.Date dateApprobation) {
        this.dateApprobation = dateApprobation;
    }

    public String getEtat() {
        return etat;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }

    public TypeVirement getType() {
        return type;
    }

    public void setType(TypeVirement type) {
        this.type = type;
    }

    // Equals and HashCode methods
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Virement virement = (Virement) o;
        return Objects.equals(id, virement.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }

    // compareTo method
    @Override
    public int compareTo(Virement virement) {
        // Implement your comparison logic here
        return 0;
    }
}
